app/src/ui/changes/commit-message.tsx

```diff
@@ -628,17 +673,26 @@ export class CommitMessage extends React.Component<
     const { isCommitting, branch: branchName, commitButtonText } = this.props
 
     const loading = isCommitting === true ? <Loading /> : undefined
-    const commitVerb = loading ? 'Committing' : 'Commit'
+
+    const isAmending = this.props.commitToAmend !== null
+    const commitVerb = isAmending
```

Comment:
This one isn't as hard to follow but.. maybe still clearer if broken into a variable for each type like below.